### DISCUSSION TITLE
Elsa 3.1 - How To get Data from Http ned point request in elsa 3.1

### DISCUSSION THREAD

**Discussion by nurhidayat-agung** [elsa-workflows/elsa-core#5526] [Q&A]
hello, 
i'm very fresh in elsa. 
i want to ask how to get data from http end point and set them to variable alse variable get and set ?
i'm following way in documentation also already import sample json but that not work 
i assumed may be any newest way to do that,
here sample json already tryied in my local
[backend-api](https://v3.elsaworkflows.io/docs/samples/http/backend-api) 
[create user](https://v3.elsaworkflows.io/docs/samples/http/post-users-api)

how to set data from http end point to variable and how to read it ?
i'm very sory if this question already answered in anoter thread 
thx

**Comment by glime-ai** [comment]
To get data from an HTTP endpoint and set it to a variable in Elsa, you can follow these steps:

1. **Create an HTTP Endpoint Activity**: This will be the trigger for your workflow. When an HTTP request is made to this endpoint, the workflow will start.

2. **Extract Data from the Request**: You can use expressions to extract data from the incoming HTTP request. For example, if you want to extract a JSON payload from a POST request, you can use an expression like `{{ Http.RequestBody }}`.

3. **Set Data to a Variable**: You can then assign the extracted data to a workflow variable using the "Set Variable" activity. You can specify the name of the variable and use the expression to set its value.

4. **Read the Variable**: Later in the workflow, you can read the variable using expressions wherever needed. For example, if you have set a variable named `userData`, you can access it using `{{ Variables.userData }}`.

Here's a simplified example based on the sample JSON you've mentioned:

```json
{
  "Name": "HTTP Endpoint Example",
  "Activities": [
    {
      "Name": "HttpEndpoint",
      "Type": "HttpEndpoint",
      "Properties": {
        "Path": "/data",
        "Method": "POST"
      }
    },
    {
      "Name": "SetVariable",
      "Type": "SetVariable",
      "Properties": {
        "VariableName": "requestData",
        "Value": "{{ Http.RequestBody }}"
      }
    },
    // ... other activities where you can use {{ Variables.requestData }}
  ]
}
```

In the above example, when a POST request is made to `/data`, the `HttpEndpoint` activity triggers the workflow. The `SetVariable` activity then sets the JSON payload from the request body to a variable named `requestData`.

If you've already imported the sample JSON workflows and they are not working, make sure that:

- The workflow is published.
- You are making the HTTP request to the correct URL (the URL of the HTTP trigger).
- Your HTTP request is correctly formatted (e.g., correct headers, method, and body).

If you continue to face issues, please provide more details about the error or behavior you're encountering, and I can assist you further.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by nurhidayat-agung** [reply]
thx @glime-ai
then how get variable in http response activities

actually i'm already try sample in documentation but not working in may 3.13 elsa maybe any way updated with new version elsa ?
